charm/src/iframe/static.ts

```diff
@@ -22,22 +65,24 @@ window.onerror = function (message, source, lineno, colno, error) {
   return false;
 };
 
-function useDoc(key) {
-  const [doc, setDoc] = React.useState(undefined);
+function useDoc(key, defaultValue = undefined) {
```

Comment:
I'll read more tomorrow - crashing tonight...

does this return defaultValue before the first useEffect finishes - or does the larger change address this issue

Reply:
larger change addresses the main issue, but it also won't attempt to overwrite or set a default state anymore - by passing the `defaultValue` the charm becomes ignorant of whether the data exists or not (which is good)